app/policies/instance_note_policy.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class InstanceNotePolicy < ApplicationPolicy
+  def create?
+    role.can?(:manage_reports)
+  end
+
+  def destroy?
+    owner? || (role.can?(:manage_federation) && role.overrides?(record.account.user_role))
+  end
```

Comment:
I'm not sure on the roles here. I think it may make sense for all moderators to be able to add notes to instances, regardless of whether or not they can manage federation; however, we don't really have a "is the user a moderator" method.

So I'm not sure what the expected roles should be for instance notes.

Reply:
Maybe at an extreme it's worth create a new role permission for `:manage_federation_notes` idk.